otherlibs/dune-rpc/private/dune_rpc_private.ml

```diff
@@ -87,28 +87,37 @@ module Where = struct
       let open Fiber.O in
       match Sys.getenv _DUNE_RPC with
       | Some d -> Fiber.return (Some (of_string d))
-      | None ->
+      | None -> (
         let of_file f =
           let+ contents = Sys.read_file f in
           Some (of_string contents)
         in
         let file = Filename.concat build_dir rpc_socket_relative_to_build_dir in
-        let rec loop file symlink_depth =
-          let* analyze = Sys.analyze_path file in
-          match analyze with
-          | `Symlink ->
-            if String.length file < 104 then
-              of_file file
-            else if symlink_depth > 0 then
-              let* link = Sys.readlink file in
-              loop link (symlink_depth - 1)
-            else
-              of_file file
-          | `Other -> Fiber.return None
-          | `Unix_socket -> Fiber.return (Some (`Unix file))
-          | `Normal_file -> of_file file
-        in
-        loop file 5
+        let* analyze = Sys.analyze_path file in
```

Comment:
I guess there are slightly bizarre things going on here (both before and after your change), but why are you changing this code, anyway?

Reply:
It was completely broken before. The point of all these hacks is to get around the 104/108 (bsd/linux) limit on unix sockets. If the path is too long, we can use resolve and pass the symlink instead. The actual socket could be created elsewhere where the path is shorter.